deploy/runtools/firesim_topology_elements.py

```diff
@@ -495,15 +585,18 @@ def run_sim_start_command(self, slotno):
         if self.uplinks:
             all_shmemportnames = [self.uplinks[0].get_global_link_id()] + [self.supernode_get_sibling_shmemportname(x) for x in range(1, num_siblings)]
 
+        assert self.server_hardware_config is not None
```

Comment:
General comment: Across all of this code I have to add these asserts so that the type checker doesn't error. These need to be analyzed much more to make sure they are in the right places (or maybe we can remove some extra `Optional` types).